thinc/model.py

```diff
@@ -462,19 +462,42 @@ def copy(self: SelfT) -> SelfT:
         layers will also be deep-copied. The copy will receive a distinct `model.id`
         value.
         """
+        return self._copy()
+
+    def _copy(self: SelfT, seen: Optional[Dict[int, Union["Model", Shim]]] = None) -> SelfT:
+        if seen is None:
+            seen = {}
```

Comment:
No, mutable default values are problematic. The `{}` object is created once when the method is defined, so if it's then modified within the method (which it is), the next method call gets the same object (now modified) as the default value rather than `{}`.

Reply:
Oh my, Python never ceases to amaze me 😱.